smithy-utils/src/main/java/software/amazon/smithy/utils/SimpleParser.java

```diff
@@ -103,6 +103,12 @@ public final int column() {
         return column;
     }
 
```

Comment:
One day we may want to try to make this support some streaming use cases. If we do, it would be better to use a marker that is set and then rewound to so the parser knows when it needs to start buffering. So instead of rewinding to an absolute position, you call mark(int numberofbytes) when you might need to backtrack, followed by reset to go back to the last set marker. This is just based on streams in Java: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/io/InputStream.html#mark(int)